runners/flink/src/main/java/org/apache/beam/runners/flink/metrics/FlinkMetricContainer.java

```diff
@@ -134,17 +159,17 @@ public void updateMetrics(String stepName, List<MonitoringInfo> monitoringInfos)
             }
           }
         });
-    updateMetrics(stepName);
+    updateMetrics();
   }
 
   /**
    * Update Flink's internal metrics ({@link this#flinkCounterCache}) with the latest metrics for a
    * given step.
    */
-  void updateMetrics(String stepName) {
```

Comment:
The "step name" had an incorrect value in portable contexts, pertaining to the name of a fused stage.

This now queries metrics for all steps that the `metricsAccumulator` has data for.

TODO: verify non-portable Flink code-paths didn't rely on this in any way.